test/apps/iodemo/io_demo.cc

```diff
@@ -650,19 +662,26 @@ class P2pDemoCommon : public UcxContext {
             }
         }
 
-        void init(size_t data_size, void *ext_buf)
+        void init(size_t data_size, void *ext_buf, bool validate)
         {
             assert(ext_buf != NULL);
             assert(_iov.empty());
             assert(_extra_buf == NULL);
 
             _data_size = data_size;
             _extra_buf = ext_buf;
+            _validate  = validate;
         }
 
         inline Buffer &operator[](size_t i) const
         {
-            return *_iov[i];
+            if (i < _iov.size()) {
+                return *_iov[i];
+            } else {
+                static Buffer dummy;
```

Comment:
global dummy buffer created by default ctor

Reply:
but what if a user does `iov[0]->buffer`? it will contain real values in case of iov, and NULL in case of `_extra_buf`

Reply:
right, it will be NULL and 0 in case of _extra buf - and recv_am_data should ignore it